### ISSUE TITLE
Bug flagged for missing property in wrong class

### ISSUE THREAD

**Issue by jlbooker** [scrutinizer-ci/php-analyzer#613]
The code in question looks like (simplified some):

```
$roommate = (<conditional>) ? $webService->foo('blah') : $webService->foo('other');
$roommate_name  = bar($roommate->last_name);
```

The analyzer is flagging this code for:
"The property `last_name` does not seem to exist in `webService`."

Which is true, the `last_name` property doesn't exist in the `webService` class, but the `$roommate` object in *not* an instance of `webService` (so I wouldn't have expected `last_name` to exist).

The `$roommate` variable holds the result of a web service call, which returns a dynamically generated object.

Perhaps the analyzer is getting tripped on the function calls nested inside of a ternary operator? It seems to be missing the fact that `$roommate` is the result of a function call on `$webService`, and not an instance of the `webService` class;

see https://scrutinizer-ci.com/g/AppStateESS/homestead/inspections/a9a1a617-ea7e-4d14-8efd-71387e32b2cc/issues/files/util/honorsInviteReport.php?status=new&selectedSeverities%5B0%5D=10&orderField=path&order=asc&issueId=f9ab4034-dcb7-45b9-9684-e7039a54f8bb
